ansible/roles/providers/tasks/azure-redhat.yml

```diff
@@ -24,6 +24,19 @@
     - name: refresh dracut
       shell: dracut -f
 
+    # https://d2iq.atlassian.net/browse/D2IQ-96417 removes extra 4 and 5 partitions so that growpart cloud-init module can expand the root filesystem to the rest of the disk.
+   - name: Remove partition number 4
```

Comment:
consider using 

```
df / --output=source
Filesystem
/dev/sda3
```
and ansible_devices

Reply:
consider scoping this down to only that azure image.

Reply:
>consider scoping this down to only that azure image.

Yes please, lets only do this for Azure.

Reply:
yeah, its already in the azure provider task, but we want to see if we could scope it down to that particular base image for rocky linux 9.1 @dlipovetsky is looking into how we can determine this using the metadata service and ansible facts

Reply:
Ansible only has an ec2 module to query the metadata service. Although the metadata service is somewhat standardized across infrastructures, I'm not sure the module works on Azure.

We have the instance metadata on disk, in `/var/run/cloud-init/instance-data.json`.

I'm now looking for the appropriate field to read. I think it's `storageProfile.imageReference`.

Reply:
We were able to do this without much complexity by checking the metadata in Ansible.